test/e2e/util_test.go

```diff
@@ -545,20 +563,50 @@ func verifyInternalIngressController(t *testing.T, name types.NamespacedName, ho
 		"--retry-delay", "20",
 		"--max-time", "10",
 	}
-	clientPod := buildCurlPod("curl-"+name.Name, echoRoute.Namespace, image, address, echoRoute.Spec.Host, extraArgs...)
+	clientPodName := types.NamespacedName{echoRoute.Namespace, "curl-" + name.Name}
+	clientPodSpec := buildCurlPod(clientPodName.Name, clientPodName.Namespace, image, address, echoRoute.Spec.Host, extraArgs...)
+	clientPod := clientPodSpec.DeepCopy()
 	if err := kclient.Create(context.TODO(), clientPod); err != nil {
-		t.Fatalf("failed to create pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+		t.Fatalf("failed to create pod %q: %v", clientPodName, err)
 	}
 	defer func() {
 		if err := kclient.Delete(context.TODO(), clientPod); err != nil {
 			if errors.IsNotFound(err) {
 				return
 			}
-			t.Fatalf("failed to delete pod %s/%s: %v", clientPod.Namespace, clientPod.Name, err)
+			t.Fatalf("failed to delete pod %q: %v", clientPodName, err)
 		}
 	}()
 
-	err = wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+	curlPodLogs := ""
+	err = wait.PollImmediate(10*time.Second, 10*time.Minute, func() (bool, error) {
+		if err := kclient.Get(context.TODO(), clientPodName, clientPod); err != nil {
+			t.Logf("error getting client pod %q: %v, retrying...", clientPodName, err)
+			return false, nil
+		}
+		// First check if client curl pod is still starting or not running.
+		if clientPod.Status.Phase == corev1.PodPending {
+			t.Logf("waiting for client pod %q to start", clientPodName)
+			return false, nil
+		} else if clientPod.Status.Phase == corev1.PodFailed {
+			// If not running, the pod most likely exited, but didn't provide a 200 response, let's try again.
+			t.Logf("client pod %q has failed...restarting", clientPodName)
+			if err := kclient.Delete(context.TODO(), clientPod); err != nil && errors.IsNotFound(err) {
+				t.Fatalf("failed to delete pod %q: %v", clientPodName, err)
+			}
+			// Wait for deletion to prevent a race condition.
+			err = kclient.Get(context.TODO(), clientPodName, clientPod)
+			for !errors.IsNotFound(err) {
```

Comment:
These inner loops should/could make use of wait.Poll* too.

Reply:
Sure okay, I debated for awhile, but I don't think adding another timebox within a timebox to further artificially restrict certain parts of the code is beneficial (I think 1 timebox is enough), but if I switched it to a `pollInfinite` (noting how you said `wait.Poll*` so hopefully that addresses your concern with style, even though it's effectively the same thing.

I confirmed the outer poll will kill the inner poll if it exceeds the outer poll's deadline.